src/Pages.tsx/PeoplePage.tsx

```diff
@@ -0,0 +1,57 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { getPeople } from '../api';
+import { PeopleTable } from '../components/PeopleTable.tsx';
+import { Person } from '../types';
+import { Loader } from '../components/Loader';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const { slug = '' } = useParams();
+  const [loading, setLoading] = useState(false);
+  const [showPeople, setShowPeople] = useState(false);
+  const [dropError, setDropError] = useState(false);
+
+  useEffect(() => {
+    setShowPeople(false);
+    setLoading(true);
+    setDropError(false);
+    const fetchPeople = async () => {
+      try {
+        const peopleFromServer = await getPeople();
+
+        setPeople(peopleFromServer);
+        setLoading(false);
+        setShowPeople(true);
+      } catch (error) {
+        setLoading(false);
+        setDropError(true);
+      }
+    };
+
+    fetchPeople();
+  }, []);
```

Comment:
It's not a good practice to write an async function in useEffect(). Create a separate function for it and call it in useEffect. Also, you can use finally in try/catch construction to set Loading as false

Reply:
why is that ? in lessons coach saying that async fn takes more code but it's easier to understand those your are requesting

Reply:
I didn't say that the async function is bad. Just said that you need to move function declaration from useEffect() and just call it in useEffect().